pkg/controlplane/http/session.go

```diff
@@ -226,6 +226,56 @@ func (s *session) DeleteAccessKeySync(deleteAccessKeyInput *v3ioc.DeleteAccessKe
 		&deleteAccessKeyInput.ControlPlaneInput)
 }
 
+// DeleteAccessKeySync deletes an access key (blocking)
+func (s *session) GetUserNameSync(getUserNameInput *v3ioc.GetUserNameInput) (*v3ioc.GetUserNameOutput, error) {
+
+	// prepare session response resource
+	userNameOutput := v3ioc.GetUserNameOutput{}
+
+	// allocate request
+	httpRequest := fasthttp.AcquireRequest()
+	defer fasthttp.ReleaseRequest(httpRequest)
+
+	responseInstance, err := s.sendRequest(getUserNameInput.ControlPlaneInput.Ctx,
+		&request{
+			method:      http.MethodGet,
+			path:        fmt.Sprintf("api/%s", "self"),
+			httpRequest: httpRequest,
+		}, getUserNameInput.ControlPlaneInput.Timeout)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// if we got cookies, set them
+	if len(responseInstance.cookies) > 0 {
```

Comment:
From what I see, the cookies are set only in `createOrUpdateResource`, in other words POST & PUT and this is something that I understand. 
However I'm not sure if we should set cookies after GET (or DELETE). It's probably ok but I'm a concerned this will introduce some regressions

Reply:
got you, will remove it